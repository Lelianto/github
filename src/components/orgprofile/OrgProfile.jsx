import React from 'react';

const OrgProfile = ({ detail }) => {
	return (
		<>
			<div className="w-full text-left">
				<div className="flex mt-5 mb-5">
					<div className="relative m-auto ml-0 mr-3 flex">
						<img className="rounded-md" width="100px" height="100px" src={detail.avatar_url} alt="" />
					</div>
					<div className="relative m-auto ml-0">
						<div id="orgName" className="text-lg font-bold text-black">
							{detail.login}
						</div>
						<div id="orgDesc" className="text-sm text-gray-500">
							{detail.description || '-'}
						</div>
					</div>
				</div>
			</div>
		</>
	)
}

export default OrgProfile;