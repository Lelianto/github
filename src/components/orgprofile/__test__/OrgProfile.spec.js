import OrgProfile from "../OrgProfile";
import { shallow } from "enzyme";
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe("Organization Profile: Check content", () => {
	let wrapper;
	let mockProps = {
		avatar_url: "https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg",
		login: "Lelianto",
		description: "React testing using jest in Organization Profile"
	}

	beforeEach(() => {
		wrapper = shallow(<OrgProfile detail={mockProps} />)
	});

	it('Renders correctly', () => {
		const tree = renderer.create(<Router><OrgProfile detail={mockProps} /></Router>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Check organization name", () => {
		const orgName = wrapper.find("#orgName")
		expect(orgName.text()).toEqual(mockProps.login)
	})

	it("Check organization description", () => {
		const orgDesc = wrapper.find("#orgDesc")
		expect(orgDesc.text()).toEqual(mockProps.description)
	})
})