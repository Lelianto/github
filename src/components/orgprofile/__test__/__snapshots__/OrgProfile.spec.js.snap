// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization Profile: Check content Renders correctly 1`] = `
<div
  className="w-full text-left"
>
  <div
    className="flex mt-5 mb-5"
  >
    <div
      className="relative m-auto ml-0 mr-3 flex"
    >
      <img
        alt=""
        className="rounded-md"
        height="100px"
        src="https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg"
        width="100px"
      />
    </div>
    <div
      className="relative m-auto ml-0"
    >
      <div
        className="text-lg font-bold text-black"
        id="orgName"
      >
        Lelianto
      </div>
      <div
        className="text-sm text-gray-500"
        id="orgDesc"
      >
        React testing using jest in Organization Profile
      </div>
    </div>
  </div>
</div>
`;
